doc/contributing/docs/markup/code.rst

```diff
@@ -1,8 +1,49 @@
 Writing about code
 ==================
 
+When writing articles, you have to format code specially, separating it from
+other text. This document will guide you through typical cases when
+it is recommended to use code highlighting.
+
+Defining what code is
+---------------------
+
+It is not difficult to recognize the code snippet in the text. First of all,
+it is probably code if the expression contains characters that ordinary words
+do not have, such as ``_, {}, [ ], .``.
+Also, you should format the expression as code if it fits at least one
+of the items in the list below:
+
+*   attribute names, values and class names,
+*   constant values for attributes,
+*   method and function names,
+*   language keywords,
+*   data types,
+*   error messages,
+*   URL for localhost and intranet,
+*   port number,
+*   package names,
+*   CLI app names.
+
+**Items we don't format as code**
+
+*   names of products, organizations and services, for example, Tarantool,
+    memtx, vinyl
+*   well-established terms such as stdin and stdout
+
+Code-blocks and inline code
+---------------------------
+
+If you have to choose between inline code and code-block highlighting,
+pay attention to the following guidelines:
+
 Code snippets
--------------
+~~~~~~~~~~~~~
+Use code-blocks when you have to highlight multiple lines of code.
+Also, use it if your code snippet contains a standalone element
+that is not a part of the article's text. In code-blocks, you can enable
+syntax highlighting if you specify the language for the snippet. Note that
```

Comment:
> In code-blocks, you can enable
syntax highlighting if you specify the language for the snippet. Note that
in code-blocks you can write comments and translate them.

These are two great and valuable ideas. However, they're very different from the topic here. The topic is whether to paste some code inline or as a block. So these ideas should go each in it's own paragraph. Probably, with examples.